dashboard/app/models/school_info.rb

```diff
@@ -83,11 +93,12 @@ def school_state=(input)
   #
   # This method reports errors if the record has a country and is invalid.
   def validate_with_country
-    return unless country
+    return unless country && should_validate?
     country == 'US' ? validate_us : validate_non_us
   end
 
   def validate_non_us
+    return unless should_validate?
```

Comment:
Ah, good catch, this is actually a bug, and should be in the validate_without_country method - it also exposes a gap in testing.